File: Leetcode/sbgo/Leetcode263-丑数.go
package sbgo

func isUgly(n int) bool {
	for n != 0 && n%2 == 0 {
		n /= 2
	}
	for n != 0 && n%3 == 0 {
		n /= 3
	}
	for n != 0 && n%5 == 0 {
		n /= 5
	}
	return n == 1
}

File: Leetcode/PY/Leetcode263-丑数.py
class Solution:
    def isUgly(self, n: int) -> bool:
        while n and n % 2 == 0:
            n /= 2
        while n and n % 3 == 0:
            n /= 3
        while n and n % 5 == 0:
            n /= 5
        return n == 1